src/lib.rs

```diff
@@ -5,52 +5,42 @@ use serde::de::DeserializeOwned;
 use serde::Serialize;
 use std::marker::Unpin;
 
-// Transport configurtatiion trait
+/// Transport configuration trait.
 pub trait TransportConfiguration<Data> {
-    // creates new transport configuration with specified network
-    // address for incoming messages listener
-    fn new(set_bind_net_addr: String) -> Self;
+    /// Creates a new transport configuration with the specified network address for incoming
+    /// message listener.
+    fn new(set_bind_net_addr: &str) -> Result<Box<Self>>;
 
-    // set bind network address for incoming messages listener
-    fn set_bind_net_addr(&mut self, address: String) -> Result<()>;
+    /// Sets the bind network address of the incoming message listener.
+    fn set_bind_net_addr(&mut self, address: &str) -> Result<()>;
```

Comment:
What are reason to change type if address from `String` to `&str`?

Reply:
I provided an explanation [here](https://github.com/Fantom-foundation/libtransport-tcp/pull/3#issuecomment-525210387).